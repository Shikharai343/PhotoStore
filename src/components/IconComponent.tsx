import React, { FC } from 'react'
import Icon from 'react-native-vector-icons/FontAwesome';

export interface IIconProps {
  name: string;
  size: number;
  color: string;
}

const IconComponent: FC<IIconProps> = (props) => {
  const {name, size, color} = props;
  return (
      <Icon name={name} size={size} color={color} />
  )
}

export default IconComponent