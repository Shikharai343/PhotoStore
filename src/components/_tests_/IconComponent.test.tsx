import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import IconComponent, { IIconProps } from '../IconComponent';

jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon')
describe('Icon', () => {
    const props: IIconProps = {
        name: 'rocket',
        size: 24,
        color: 'white'
    };

    const wrapper: ShallowWrapper = shallow(
        <IconComponent {...props} />
    );

    it('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});