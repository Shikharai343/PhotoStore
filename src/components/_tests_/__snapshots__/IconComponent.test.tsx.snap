// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot 1`] = `
<Icon
  color="white"
  name="rocket"
  size={24}
/>
`;
